tests/pytests/test_cursors.py

```diff
@@ -189,35 +189,35 @@ def testNumericCursor(env):
     env.assertEqual(cursor, 0)
 
 
-def testIndexDropWhileIdle(env):
+def testIndexDropWhileIdle(env: Env):
     conn = getConnectionByEnv(env)
 
     env.expect('FT.CREATE idx SCHEMA t numeric').ok()
 
-    num_docs = 3
+    num_docs = 6
     for i in range(num_docs):
         conn.execute_command('HSET', f'doc{i}' ,'t', i)
 
-    count = 1
-    res, cursor = conn.execute_command('FT.AGGREGATE', 'idx', '*', 'WITHCURSOR', 'COUNT', count)
+    count = 3
+    res, cursor = env.cmd('FT.AGGREGATE', 'idx', '*', 'WITHCURSOR', 'COUNT', count)
 
     # Results length should equal the requested count + additional field for the number of results
-    # (which is meaningless is ft.aggregate)
-    env.assertEqual(len(res), count + 1)
+    # (which is meaningless with ft.aggregate)
+    env.assertEqual(res[1:], [[]] * count, message=f'res == {res}')
 
     # drop the index while the cursor is idle/ running in bg
-    conn.execute_command('ft.drop', 'idx')
+    env.expect('FT.DROPINDEX', 'idx').ok()
```

Comment:
what's the difference? is ft.drop deprecated? (couldn't find it in the docs)

Reply:
The test was flaky, most likely because the `env` connection and `conn` were connected to different shards.
We remove the flakiness by using the `env` connection only for search commands. If this test becomes flaky again, we will have better logs now.

Regarding the drop command, yes `FT.DROP` is deprecated. The difference is that `DROP` drops the documents as well by default and `DROPINDEX` does not (but can if it is given the `DD` flag)